### PR TITLE
MM-38366 - trial started carousel benefits modal

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds improvements to the start trial modal by showing another modal with all the benefits once the user clicks the option to start the trial.

* Pending unit tests to benefits modal and carousel component.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38366

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/9149

#### Screenshots

https://user-images.githubusercontent.com/10082627/137567608-b44ee066-34cb-4f1d-9acb-c04bff9211ce.mp4

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by marianunez** [comment]
> Not urgent - what would be the best place for me to try it out? I tried the above test server, but it's for a Cloud environment instead of Self-Hosted?

This should work with the `Setup Cloud Test Server` tag (instead of the CWS one)

**Comment by pablovelezvidal** [comment]
> Looks like we haven't yet added the "Learn More" CTA in the main menu, is that coming in a separate PR?
> 
> image
> 
> https://www.figma.com/file/E2sh7Ny5wlyDhqpgbwowxm/Enterprise-changes?node-id=0%3A1

hey @jasonblais , yes, it is implemented as part as this change, but there is a problem with the current implementation. I will push asap a fix for it, I have already a fix for it locally.


Captura de pantalla 2021-10-19 a las 12 17 25


Captura de pantalla 2021-10-19 a las 12 17 46


Captura de pantalla 2021-10-19 a las 12 18 13

File: components/widgets/menu/menu_items/menu_start_trial.tsx [hunk 1]
```diff
@@ -30,16 +35,26 @@ const MenuStartTrial = (props: Props): JSX.Element | null => {
...
+    const license = useSelector(getLicense);
     const isPrevLicensed = prevTrialLicense?.IsLicensed;
-    const isCurrentLicensed = currentLicense?.IsLicensed;
+    const isCurrentLicensed = license?.IsLicensed;
+
+    const isCurrentLicenseTrial = isTrialLicense(license);
 
     // Show this CTA if the instance is currently not licensed and has never had a trial license loaded before
-    const show = (isCurrentLicensed === 'false') && (isPrevLicensed === 'false');
+    const show = (isCurrentLicensed === 'false' && isPrevLicensed === 'false') || isCurrentLicenseTrial;
```
Comment (AGMETEOR) [inline components/widgets/menu/menu_items/menu_start_trial.tsx:57]: Hello @pablovelezvidal , did we want to show this CTA if current license is a trial

Comment (pablovelezvidal) [reply components/widgets/menu/menu_items/menu_start_trial.tsx:57]: yes!

Comment (pablovelezvidal) [reply components/widgets/menu/menu_items/menu_start_trial.tsx:57]: yes, and it shows a different modal content. Check line 119

**Pull request merged**